File: LeetCode problem set/66_Plus_One.js
// https://leetcode.com/problems/plus-one/
const plusOne = (digits) => {
  let idx = digits.length - 1;
  for (let i = 0; i < digits.length; i++) {
    if (digits[idx] === 9) {
      digits[idx] = 0;
      idx--;
    } else {
      digits[idx]++;
      return digits;
    }
  }
  digits.unshift(1);
  return digits;
};

const data = plusOne([3, 9, 9]);
data.forEach((item) => console.log(item));

File: LeetCode problem set/66_Plus_One.cpp
// https://leetcode.com/problems/plus-one/

#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
      int idx = digits.size() - 1;
      for (int i = 0; i < digits.size(); i++) {
        if(digits[idx] == 9) {
          digits[idx] = 0;
          idx--;
        } else {
          digits[idx] += 1;
          return digits;
        }
      }
      digits.insert(digits.begin(), 1);
      return digits;
    }
};

int main(){
  Solution s;
  vector<int> digits = {1,9,9};
  vector<int> res = s.plusOne(digits);
  for (int i = 0; i < res.size(); i++) {
    cout << res[i] << " ";
  }
  return 0;
}